openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1952,3 +1952,28 @@ def import_blocks_create_task(library_key, course_key, use_course_key_as_block_i
     log.info(f"Import block task created: import_task={import_task} "
              f"celery_task={result.id}")
     return import_task
+
+
+def create_or_update_xblock_upstream_link(xblock, course_key: str, course_name: str, created: datetime | None = None):
+    """
+    Create or update upstream->downstream link in database for given xblock.
+    """
+    if not xblock.upstream:
+        log.info(f"No upstream found for xblock: {xblock.usage_key}")
+        return None
+    upstream_usage_key = UsageKeyV2.from_string(xblock.upstream)
+    try:
+        lib_component = get_component_from_usage_key(upstream_usage_key)
+    except ObjectDoesNotExist:
+        log.exception("Library block not found!")
+        lib_component = None
+    authoring_api.update_or_create_entity_link(
```

Comment:
If we're returning `None` when we do nothing, shouldn't this return something truthy?

Reply:
It is possible that a course is imported which has an xblock that an upstream which does not exist in this instance. We want to still save them and link them when the library block is imported (this is not implemented yet, I think I'll do it as part of a future task).